### PR TITLE
Hotfix detector.py

### PR DESCRIPTION
**Pull request by fschlueter**
I did not run the code but just from looking at it I thought this was wrong? @sjoerd-bouma would you mind double reading this?

### CODE REVIEW

**Comment by fschlueter** [comment]
For the first commit in this PR I have an extra PR #774

File: NuRadioReco/detector/detector.py [hunk 1]
```diff
@@ -57,7 +57,7 @@ def Detector(*args, **kwargs):
             - kwargs["source"] == "rnog_mongo" -> `NuRadioReco.detector.RNO_G.rnog_detector`
             - kwargs["source"] == "sql" -> `NuRadioReco.detector.detector_base`
             - kwargs["source"] == "json" or "dictionary" -> `NuRadioReco.detector.detector_base` or
-              `NuRadioReco.detector.generic_detector`
+            `NuRadioReco.detector.generic_detector`
```
Comment (sjoerd-bouma) [inline NuRadioReco/detector/detector.py:60]: bad rst formatting (previous was correct)

**Review by sjoerd-bouma — COMMENTED** [review]
As far as I can tell, this PR wouldn't actually change anything? If `source=='dictionary'`, `station_dict` and `dictionary` are the same, so nothing changes. And if `source=='json'`, the `dictionary` argument is ignored (as it should be), so the fact that you now pass a dictionary instead of `None`, while arguably not desirable, doesn't change anything.

I would probably recommend closing this. `station_dict` was originally (and still is) used to check whether the detector is generic or not.

File: NuRadioReco/detector/detector.py (hunk 1 above)
Comment (fschlueter) [reply NuRadioReco/detector/detector.py:60]: okay, I changed it because my editor complained ...

**Comment by fschlueter** [comment]
I missed the part that the source=="json" is correctly handled within the classes

**Pull request closed without merging**